### DISCUSSION TITLE
How to use Assign values to variables in Send Request

### DISCUSSION THREAD

**Discussion by eamell** [postmanlabs/postman-flows#259] [Q&A] [answered]
I have a request with an example response.  I am using it in a flow and I want to pull data out of the response and assign it to a variable.  I see the that assign variable flow element is deprecated in favor of using the _assign values to variables_ option in send request.  The problem I am having is that while the _Assign Variables_ flow element is working fine the similar option on _Send Request_ is not.

This is what I am seeing in Send Request:

![image](https://user-images.githubusercontent.com/3307909/154987519-62a88270-8ffc-457d-887a-001db87aa1d0.png)

And this is what I see in Assign Variables:

![image](https://user-images.githubusercontent.com/3307909/154987645-b8e95d2d-0051-4c13-aaea-b505b5202cfe.png)

As you can see Assign Variables shows me the response object per my example while Send Request doesn't.  I am not sure if I am doing something wrong or if I uncovered an issue (which is fine given it is a beta).

**Comment by samay-v** [comment, marked as answer]
Hey @eamell 
Both of these options show you the data that comes in from the input port. In the Assign Variables block, you see the response body that you have piped in through the input. 

In the send request block also, it shows provides the option to assign data received as input to the variables defined in the Request. 

If you wish to use the response in another request, I'd suggest you connect the `Response` port of the 1st request to the `variables` port of the second request and assign the value in the second request.

**Reply by eamell** [reply]
Thank you @samay-v!  I was thinking of it differently; rather than assigning the variables from the data coming in I was thinking I should be assigning variables on the data going out.  In retrospect this approach makes more sense because it allows the request to control what it does with the data coming in.  I really like Flows but it does need some additional documentation.

**Reply by samay-v** [reply]
Thanks @eamell! Glad you like flows
We're working on adding documentation in the [Postman learning center](https://learning.postman.com/). Till then we maintain some documentation [here](https://www.postmanlabs.com/postman-flows/). Let me know if that helps 👍🏽
